wled00/wled.cpp

```diff
@@ -603,6 +603,7 @@ void WLED::initConnection()
   ws.onEvent(wsEvent);
   #endif
 
+  if (wifiDisabled) return; // can still use ethernet
```

Comment:
Please put this between `#ifdef USERMOD_BLE_2_JSON` ... `#endif` --> see my comment on wled.h.

Reply:
> Please put this between `#ifdef USERMOD_BLE_2_JSON` ... `#endif` --> see my comment on wled.h.

I'm working on 3 usermods that require this sort of thing, one of which is this exact feature set, bluetooth json api, by means of a different method.
This PR beat me to the submission. So I will say that locking this behind a single usermod define would be troublesome for other usermods.

Reply:
> I'm not sure if disabling Wifi should be part of the "standard WLED", as it could (possibly) render the device unusable (no more access to web UI, no AP could be opened), and a factory reset would be needed to "revive" WLED? Or is there an existing "kill switch" already to shut down BLE and start Wifi again?
> 
> @blazoncek what would you recommend? Maybe create a new flag like `WLED_SUPPORT_WIFI_OFF` ?

My current implementation has serial overrides when the option is compiled in, and the usermod that uses these features should have additional overrides. Getting locked out of wifi should never be an option.
I was going to put in a bluetooth serial JSON api usermod in today, but then I saw this.

Reply:
Even though BLE could substitute WiFi connection (in theory) none of the underlying libraries are up to it AFAIK.
So unless the full push/pull API for BLE is implemented somewhere in addition to wled_server.cpp for WiFi, disabling WiFi should not be done in usermod but should better be a standard WLED setting.

Though, if properly implemented I have no objections if it is done within usermod.

Reply:
i'm not sure where this leaves this part of the code... if you are suggesting there should be a standard wled config for shutting off wifi, I would argue against that as that flag (without this mod or one like it) would render the device un-reachable. with this mod (and other potential mods), one can then turn back on wifi through a ble call to set the state like this:

```
{"bleToggle":true}
```

I think maybe we do as @ChuckMash suggests and use `WLED_SUPPORT_WIFI_OFF` ... then we can state in the readme to make sure to define that flag in the build. keep in mind, they way i've implemented it here, you can only turn wifi off at `setup` time... so the mod has to be functioning at boot to set wifi off... if you install another version of wled without the mod, the wifi will go back to normal (which is what we'd want)